lms/views/dashboard/api/user.py

```diff
@@ -203,9 +203,51 @@ def _students_query(
         segment_authority_provided_ids: list[str],
         h_userids: list[str] | None = None,
     ):
+        course_ids = self.request.parsed_params.get("course_ids")
+        # Single assigment fetch
+        if (
```

Comment:
We have a tradeoff in this PR, we get this conditionals to pick the right query to use in exchange for simpler queries.